### ISSUE TITLE
Integer division by zero in inspector

### ISSUE THREAD

**Issue by alex-ilin** [factor/factor#2536]
Fresh bootstrap of commit `bac5f95f`. Command: `all-instances inspector`

![factor_WGzmbWfTh7](https://user-images.githubusercontent.com/169039/147882628-85f83398-030c-444f-b31c-67e4c2f69c32.png)
The same exception can be seen if one clicks the array in the listener stack to open the GUI Inspector window.

**Comment by erg** [comment]
Currently division by zero for ratios looks like an unprintable error on windows:

```
0 0 ratio boa
~pprint error: ratio~
```

We could fix it to be like ``3/0``, however, these ratios won't round-trip because ``3/0`` isn't valid to the parser. (also what would it do, immediately throw the division by zero exception?)
```
M: ratio >base
    [ >fraction dup zero? [ 0 -rot ] [ [ /mod ] keep ] if ]
    [ [ >base ] curry tri@ ] bi*
    "/" glue over first-unsafe {
        { CHAR: 0 [ nip ] }
        { CHAR: - [ append ] }
        [ drop "+" glue ]
    } case ;

! with this patch...
3 0 ratio boa 10 >base .
"3/0"
```

And your ``all-instances inspector`` code takes forever.

**Comment by mrjbq7** [comment]
```factor
\ ratio props>> "prototype" of
```

**Comment by mrjbq7** [comment]
I think fixed by 37c017617e60e14d3f86cffbea5cb270a3ad2a4f and 22f07267f995c1f0ba4dda32cecbfffb6114a234.
